src/Lachain.Consensus/AbstractProtocol.cs

```diff
@@ -134,7 +137,7 @@ public void Start()
                 catch (Exception e)
                 {
                     Logger.LogError($"{Id}: exception occured while processing message: {e}");
-                    Terminated = true;
+                    Terminate();
```

Comment:
I don't sure it is correct,  we can get deadlock here because Terminate contains lock(_queueLock), so,  if we get  exception from inside other lock Terminate will never be executed really

Reply:
understood, reverting change